File: jvPY/2573.py
# No me olvides
# https://jv.umsa.bo/problem.php?id=2573

n = int(input())
aux = ""
for _ in range(n):
    cad = input().strip(" ") +"X"
    for i in range(len(cad)-1):
        if cad[i].isnumeric():
            aux+=cad[i]
            if not cad[i+1].isnumeric():
                aux+=" "
res = aux[:-1].split(" ")
res = [int(x) for x in res]
res.sort()
print(*res)

    

File: jvCPP/2573.cpp
// No me olvides
// https://jv.umsa.bo/problem.php?id=2573

#include <iostream>
#include <vector>
#include <algorithm>
#include <sstream>
using namespace std;

int main() {
    int n;
    cin >> n;
    cin.ignore();
    string aux;
    for (int i = 0; i < n; ++i) {
        string cad;
        getline(cin >> ws, cad);
        cad = cad + "X";
        for (int j = 0; j < cad.length() - 1; ++j) {
            if (isdigit(cad[j])) {
                aux += cad[j];
                if (!isdigit(cad[j + 1])) {
                    aux += " ";
                }
            }
        }
    }

    istringstream iss(aux);
    vector<int> res;
    int num;
    while (iss >> num) {
        res.push_back(num);
    }

    sort(res.begin(), res.end());

    string ans = "";
    for (const auto& x : res) {
        ans += to_string(x) + " ";
    }
    ans.pop_back();
    cout << ans <<endl;

    return 0;
}
